sinks/kafka/driver.go

```diff
@@ -0,0 +1,166 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kafka
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/golang/glog"
+	"github.com/optiopay/kafka"
+	"github.com/optiopay/kafka/proto"
+	"k8s.io/heapster/extpoints"
+	sink_api "k8s.io/heapster/sinks/api"
+	kube_api "k8s.io/kubernetes/pkg/api"
+	"net/url"
+	"time"
+)
+
+const (
+	partition                int32         = 0
+	brokerClientID           string        = "kafka-sink"
+	brokerDialTimeout        time.Duration = 10 * time.Second
+	brokerDialRetryLimit     int           = 10
+	brokerDialRetryWait      time.Duration = 500 * time.Millisecond
+	brokerAllowTopicCreation bool          = true
+	brokerLeaderRetryLimit   int           = 10
+	brokerLeaderRetryWait    time.Duration = 500 * time.Millisecond
```

Comment:
Lots of interesting stuff at blog.golang.org :) https://blog.golang.org/constants